### PR TITLE
fix Vbft Block serdes

### PR DESCRIPTION
**Pull request by Honglei-Cong**

### CODE REVIEW

**Comment by Honglei-Cong** [comment]
fix #990

**Review by xiemylogos — DISMISSED** [review]

File: consensus/vbft/types.go [hunk 1]
```diff
@@ -109,22 +109,29 @@ func (blk *Block) Deserialize(data []byte) error {
...
-				emptyBlock = block2
+	hasEmptyBlock, irr, eof := source.NextBool()
+	if irr {
+		return fmt.Errorf("read empty-block-bool: %s", common.ErrIrregularData)
+	}
+	if eof {
+		return fmt.Errorf("read empty-block-bool: %s", io.ErrUnexpectedEOF)
+	}
+	if hasEmptyBlock {
+		if source.Len() > 0 {
```
Comment (laizy) [inline consensus/vbft/types.go]: remove this check.

File: consensus/vbft/types.go [hunk 2]
```diff
@@ -109,22 +109,29 @@ func (blk *Block) Deserialize(data []byte) error {
...
+			buf2, _, irregular, eof := source.NextVarBytes()
+			if irregular == false && eof == false {
+				block2, err := types.BlockFromRawBytes(buf2)
+				if err == nil {
+					emptyBlock = block2
+				}
```
Comment (laizy) [inline consensus/vbft/types.go]: err != nil case is allowed?

Comment (Honglei-Cong) [reply consensus/vbft/types.go]: fixed

File: consensus/vbft/types.go (hunk 1 above)
Comment (Honglei-Cong) [reply consensus/vbft/types.go]: fixed

**Review by laizy — APPROVED** [review]

**Pull request merged**